### DISCUSSION TITLE
initializing item services inside the routes vs module level

### DISCUSSION THREAD

**Discussion by Nisthar** [directus/directus#13169] [Archive]
Whats the difference between the two?

![image](https://user-images.githubusercontent.com/11994157/167266704-768db7b0-04e8-413d-b1c6-17e807263cc7.png)
 
 vs 
 
 initializing it inside every routes.

**Comment by rijkvanzanten** [comment]
The schema can change between requests, so it's safer to initialize the ItemsService in the route. You can then use `req.schema` instead of using `getSchema()` as well 👍🏻

**Reply by Nisthar** [reply]
> The schema can change between requests, so it's safer to initialize the ItemsService in the route. You can then use `req.schema` instead of using `getSchema()` as well 👍🏻

also i noticed that itemsService works fine without passing the database object like
```js
    const usersService = new ItemsService("directus_users", {
      schema: req.schema,
    });
``` 

what's the difference?

**Reply by rijkvanzanten** [reply]
For endpoints, there isn't any difference. For hooks, the passed `database` object is either the current database transaction or the global database connection, so it's important there.
